app/controllers/users_controller.rb

```diff
@@ -12,12 +12,13 @@ class UsersController < ApplicationController
 
   def create
     user_params = params.require(:user).permit(*WHITELISTED_ATTRIBUTES)
-    user = User.new user_params
-    user.rosters << @roster
-    if user.save
-      confirm_change(user)
+    membership_params = user_params[:membership]
+    user_params = parse_roster_ids(user_params.except(:membership))
+    @user = User.new(user_params)
```

Comment:
Why does `@user` need to be an instance variable?

Reply:
Because we called a method that uses the instance variable.

Reply:
Should be a local variable.

Reply:
No, @avacmeyer is correct. [The update membership method](https://github.com/umts/screaming-dinosaur/blob/ef61b663e6a61f224bc3d8888338988b2ce55264/app/controllers/users_controller.rb#L68-L74) uses an instance variable.

Reply:
oh weird ok